ravenframework/Optimizers/RavenSampled.py

```diff
@@ -334,36 +351,64 @@ def finalizeSampler(self, failedRuns):
+    # sanity check: if there's no history (we never got any answers) then report rather than crash
     if len(self._optPointHistory[traj]) == 0:
       self.raiseAnError(RuntimeError, f'There is no optimization history for traj {traj}! ' +
                         'Perhaps the Model failed?')
-    opt = self._optPointHistory[traj][-1][0]
-    val = opt[self._objectiveVar]
-    self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
-    if bestValue is None or val < bestValue:
-      bestValue = val
-      bestTraj = traj
-    bestOpt = self.denormalizeData(self._optPointHistory[bestTraj][-1][0])
-    bestPoint = dict((var, bestOpt[var]) for var in self.toBeSampled)
-    self.raiseADebug('')
-    self.raiseAMessage(' - Final Optimal Point:')
-    finalTemplate = '    {name:^20s}  {value: 1.3e}'
-    finalTemplateInt = '    {name:^20s}  {value: 3d}'
-    self.raiseAMessage(finalTemplate.format(name=self._objectiveVar, value=s * bestValue))
-    self.raiseAMessage(finalTemplateInt.format(name='trajID', value=bestTraj))
-    for var, val in bestPoint.items():
-      self.raiseAMessage(finalTemplate.format(name=var, value=val))
-    self.raiseAMessage('*' * 80)
-    # write final best solution to soln export
-    self._updateSolutionExport(bestTraj, self.normalizeData(bestOpt), 'final', 'None')
+
+    if not self._isMultiObjective:
+      opt = self._optPointHistory[traj][-1][0]
+      val = opt[self._objectiveVar[0]]
+      self.raiseADebug(statusTemplate.format(status='active', traj=traj, val=s * val))
+      if bestValue is None or val < bestValue:
+        bestValue = val
+        bestTraj = traj
+      bestOpt = self.denormalizeData(self._optPointHistory[bestTraj][-1][0])
+      bestPoint = dict((var, bestOpt[var]) for var in self.toBeSampled)
+      self.raiseADebug('')
+      self.raiseAMessage(' - Final Optimal Point:')
+      finalTemplate = '    {name:^20s}  {value: 1.3e}'
+      finalTemplateInt = '    {name:^20s}  {value: 3d}'
+      self.raiseAMessage(finalTemplate.format(name=self._objectiveVar[0], value=s * bestValue))
+      self.raiseAMessage(finalTemplateInt.format(name='trajID', value=bestTraj))
+      for var, val in bestPoint.items():
+        self.raiseAMessage(finalTemplate.format(name=var, value=val))
+      self.raiseAMessage('*' * 80)
+      # write final best solution to soln export
+      self._updateSolutionExport(bestTraj, self.normalizeData(bestOpt), 'final', 'None')
+    else: #self._isMultiObjective true
+      for i in range(len(self._optPointHistory[traj][-1][0][self._objectiveVar[0]])):
+        opt = self._optPointHistory[traj][-1][0]
+        key = list(opt.keys())
+        val = [item[i] for item in opt.values()]
+        optElm = {key[a]: val[a] for a in range(len(key))}
+        optVal = [(-1*(self._minMax[b]=='max')+(self._minMax[b]=='min'))*optElm[self._objectiveVar[b]] for b in range(len(self._objectiveVar))]
+
```

Comment:
@joshua-cogliati-inl, is the line optVal ever used? Should it be there? I think we need to check the multiplication by -1 in case of maximization. I think we shouldn't have had to deal with that in tha GA and it is taken care of in the parent classes, but I recall that was not the case, I removed several attempts but in RAVENSampled, I think it should be handeled some where. Is it possible to check if there are any redunduncies here. Having correct answers doesn't prove it is correct, multiplying by -1, four times gives the same result as two times. Can you help us find any unnecessary ones?